### ISSUE TITLE
axe-core fails if Array.prototype.include is extended

### ISSUE THREAD

**Issue by Lausselloic** [dequelabs/axe-core#449]
I'm going to became crazy, I've errors with the chrome extension (axe-dev-tool 3.0.0 - axe-core 2.3.1 - chrome 59 - on windows 7)
On the url : http://www.orange.pl/telefonia_komorkowa_indywidualny.phtml
The extension throw accessibility failures, but when I execute the following in console, result object is empty : 
var script = document.createElement('script');
script.type = 'text/javascript';
script.src = 'https://cdnjs.cloudflare.com/ajax/libs/axe-core/2.3.1/axe.js';
document.head.appendChild(script);
setTimeout(function(){
  axe.run(function(err, results) {
    if (err) throw err;
    console.log(results);
  });
}, 5000);

I've tried this console script on Firefox (50.1.0) and result isn't empty, but Firefox crash or freeze each time.

I've tried to disable Iframes and some other rules... Without success.

Any idea?

**Comment by dylanb** [comment]
The difference between executing something in the console, and executing something in the extension is the fact that the extension executes its JavaScript in a sandbox that separates the JavaScript execution environment from the page's JavaScript.

If that page is doing something funky to mess with the standard JavaScript library, then that would explain the difference. The fact that the extension works properly indicates that it is probably a problem with the page.

**Comment by Lausselloic** [comment]
Yes for sure. After more investigation, if I block the script http://www.orange.pl/ocp-http/170717/map/js/merged/ptk-infoportal-common.js with a proxy, aXe works well.

Quite strange this script is a merge of prototype and Dojo. I will investigate more later :-) I love to understand what happen

**Comment by dylanb** [comment]
Interesting - if you find any details that could help us to improve the robustness of axe-core - please share them

**Comment by Lausselloic** [comment]
Yes don't worry I will let you know :-)

**Comment by WilcoFiers** [comment]
I've found the problem. There is a `.include()` method on `Array.prototype` which is throwing aXe off. Axe is seeing that and assuming you don't want to run any rule.

@dylanb I suggest we keep this open. Our ruleShouldRun() utility should be secured better from accidentally overloading things.

**Comment by Lausselloic** [comment]
Good catch up! I was also testing with a functional page, and injecting prototype before axe, create any problem.

Let me know if this is fixed in the core. Thanks for the quick reply
